examples/algorithms/cuda/include/cuda/track_finding/seeding_algorithm.hpp

```diff
@@ -22,7 +22,9 @@ class seeding_algorithm {
     using output_type =
         std::pair<host_internal_spacepoint_container, host_seed_container>;
 
-    seeding_algorithm(vecmem::memory_resource* mr = nullptr) : m_mr(mr) {
+    seeding_algorithm(vecmem::memory_resource* mr)
```

Comment:
I guess this is not the only place where this pattern shows up, but this is where I'll comment about it...

We should only pass the memory resources around by pointer if it is allowed to pass a null pointer. In this case it's not really allowed anymore. So it would make for a better interface (in my opinion) to receive the memory resource as a reference. (`vecmem::memory_resource&`)

It is indeed annoying that the std classes receive memory resources through pointers. Since they do allow the pointers to be invalid. (In which case they fall back on a default resource.) But with vecmem in most cases we just need to pass explicitly created memory resources around. (Otherwise compilation on macOS can fail. As the Apple Clang compiler doesn't handle "default memory resources" quite correctly. Which provides a pretty good debugging tool for vecmem in the end. :stuck_out_tongue:)

Reply:
Okay!

Reply:
Hey wait. If the input is reference, it's not polymorphic anymore for different `memory_resource`.

Reply:
Do you just want it to be `(vecmem::cuda::memory_resource& mr)` for cuda algorithm and `(vecmem::sycl::memory_resource& mr)` for sycl algorithm?

Reply:
References in C++ are the same in terms of polymorpism to pointers, so this should not be a problem. :slightly_smiling_face:

Reply:
OK thanks. The changes are applied